go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/containerd/containerd
 
-go 1.23.0
+go 1.24.0
```

Comment:
There was an earlier discussion on changing the minimum go version on an existing release branch and we didnt want to do this bump cc: @thaJeztah

Reply:
cc @chrishenzie @samuelkarp @dmcgowan

Reply:
Discussed with Mike on this, updating the minimum version will be the best way forward.